File: ios/MenCare/Core/Data/Fake/FakeTimeBlockRepository.swift
import Foundation

@MainActor
final class FakeTimeBlockRepository: TimeBlockRepository {

    private let store: InMemoryStore

    init(store: InMemoryStore) {
        self.store = store
    }

    func blocksForOperator(_ operatorId: String?, date: LocalDate) async throws -> [TimeBlock] {
        store.timeBlocks
            .filter { (operatorId == nil || $0.operatorId == operatorId) && $0.date == date }
            .sorted { $0.range.start < $1.range.start }
    }

    func blocksForWeek(_ weekStart: LocalDate, operatorId: String?) async throws -> [TimeBlock] {
        let weekEnd = weekStart.plusDays(7)
        return store.timeBlocks.filter {
            (operatorId == nil || $0.operatorId == operatorId) && $0.date >= weekStart && $0.date < weekEnd
        }
    }

    func upcomingBlocks(from: LocalDate, weeks: Int) async throws -> [TimeBlock] {
        let horizon = from.plusWeeks(max(weeks, 1))
        return store.timeBlocks
            .filter { $0.date >= from && $0.date < horizon }
            .sorted { ($0.date, $0.range.start) < ($1.date, $1.range.start) }
    }

    func conflictsFor(_ block: TimeBlock) async throws -> [Appointment] {
        store.appointments.filter {
            $0.operatorId == block.operatorId && $0.date == block.date && $0.isActive &&
                block.range.overlaps(TimeRange($0.time, $0.end.time))
        }
    }

    func createBlock(_ block: TimeBlock) async -> AppResult<TimeBlock> {
        if !((try? await conflictsFor(block)) ?? []).isEmpty {
            return .failure(.validation(field: "conflicts"))
        }
        var saved = block
        if saved.id.isEmpty {
            saved = TimeBlock(
                id: store.newId("blk"), operatorId: block.operatorId, reason: block.reason,
                date: block.date, range: block.range, label: block.label
            )
        }
        store.timeBlocks.append(saved)
        return .success(saved)
    }

    func deleteBlock(_ blockId: String) async -> AppResult<Void> {
        store.timeBlocks.removeAll { $0.id == blockId }
        return .success(())
    }
}

File: android/core/data/src/main/kotlin/com/devora/mencare/core/data/fake/FakeTimeBlockRepository.kt
package com.devora.mencare.core.data.fake

import com.devora.mencare.core.common.AppError
import com.devora.mencare.core.common.AppResult
import com.devora.mencare.core.data.repository.TimeBlockRepository
import com.devora.mencare.core.model.Appointment
import com.devora.mencare.core.model.TimeBlock
import com.devora.mencare.core.model.TimeRange
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map
import java.time.LocalDate
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class FakeTimeBlockRepository @Inject constructor(
    private val store: InMemoryStore,
) : TimeBlockRepository {

    override fun blocksForOperator(operatorId: String, date: LocalDate): Flow<List<TimeBlock>> =
        store.timeBlocks.map { list ->
            list.filter { it.operatorId == operatorId && it.date == date }.sortedBy { it.range.start }
        }

    override fun blocksForWeek(weekStart: LocalDate): Flow<List<TimeBlock>> =
        store.timeBlocks.map { list ->
            val weekEnd = weekStart.plusDays(7)
            list.filter { it.date >= weekStart && it.date < weekEnd }
        }

    override fun upcomingBlocks(from: LocalDate): Flow<List<TimeBlock>> =
        store.timeBlocks.map { list ->
            list.filter { it.date >= from }.sortedWith(compareBy({ it.date }, { it.range.start }))
        }

    override suspend fun conflictsFor(block: TimeBlock): List<Appointment> =
        store.appointments.value.filter {
            it.operatorId == block.operatorId && it.date == block.date && it.isActive &&
                block.range.overlaps(TimeRange(it.time, it.end.toLocalTime()))
        }

    override suspend fun createBlock(block: TimeBlock): AppResult<TimeBlock> {
        if (conflictsFor(block).isNotEmpty()) {
            return AppResult.Failure(AppError.Validation("conflicts"))
        }
        val saved = if (block.id.isBlank()) block.copy(id = store.newId("blk")) else block
        store.timeBlocks.value += saved
        return AppResult.Success(saved)
    }

    override suspend fun deleteBlock(blockId: String): AppResult<Unit> {
        store.timeBlocks.value = store.timeBlocks.value.filterNot { it.id == blockId }
        return AppResult.Success(Unit)
    }
}
